contracts/CliptoToken.sol

```diff
@@ -0,0 +1,85 @@
+// SPDX-License-Identifier: AGPL-3.0 License
+pragma solidity ^0.8.10;
+
+import "@openzeppelin/contracts-upgradeable/token/ERC721/ERC721Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "@openzeppelin/contracts/utils/Counters.sol";
+
+import "./CliptoTokenStorage.sol";
+
+contract CliptoToken is CliptoTokenStorage, Initializable, ERC721Upgradeable {
+    using Counters for Counters.Counter;
+
+    Counters.Counter private _currentTokenId;
+
+    mapping(uint256 => string) private _tokenURIs;
```

Comment:
If this is going to be upgradable, state variables within the contract will cause major headaches later down the line.